routes/users.go

```diff
@@ -34,6 +35,20 @@ type registerRequest struct {
 	DateOfBirth *time.Time `json:"dateOfBirth"`
 }
 
+type updateUserRequest struct {
+	PasswordHash    []byte
+	NameFirst       *string    `json:"firstName"`
+	NameLast        *string    `json:"lastName"`
+	Email           *string    `json:"email"`
+	PhoneNumber     *string    `json:"phoneNumber"`
+	City            *string    `json:"city"`
+	State           *string    `json:"state"`
+	DateOfBirth     *time.Time `json:"dateOfBirth"`
```

Comment:
I believe time.Time can already be nil

Reply:
tried and got error with "cannot convert nil to type time.Time". will keep as a pointer